tests/AgDatabaseMove.Unit/SmoFacadeListenerTest.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Collections.Generic;
+using Xunit;
+
+namespace AgDatabaseMove.Unit
+{
+  public class SmoFacadeListenerTest
+  {
+
+    /// <summary>
+    /// Tests for Listener.SplitDomainAndPort()
+    /// </summary>
+
+    private const string DOMAIN = "abc.def.ghi";
+    private const string HOST = "abc";
+
+    // {input, expectedDomain, expectedPort (can be null)}
+    public static IEnumerable<object[]> ValidPorts => new List<object[]> {
+      new object[] { $"{DOMAIN}", DOMAIN },
+      new object[] { $"{DOMAIN},123", DOMAIN, ",123" },
+      new object[] { $"{DOMAIN}\\SQL", DOMAIN, "\\SQL" },
+
+      new object[] { $"{HOST}", HOST },
+      new object[] { $"{HOST},123", HOST, ",123" },
+      new object[] { $"{HOST}\\SQL", HOST, "\\SQL" }
```

Comment:
No need to interpolate $"{DOMAIN}" and $"{HOST} when it's already a string and not being concatenated.
Some other strings like ",123" and "\\SQL" could be put in const string variables too